apps/server/src/app/reporting/opensearch.service.ts

```diff
@@ -24,13 +28,19 @@ export class OpenSearchService {
   ) {
     const reportId = randomUUID();
 
-    const { provider, type, properties, metadata, request, response } = dto;
+    if (dto.provider !== ProviderType.OpenAI) {
+      throw new InternalServerErrorException("Unsupported provider");
+    }
+
+    const { provider, type, properties, metadata, request, response } =
+      dto as ReportRequestDto<ProviderType.OpenAI>;
```

Comment:
Once we introduce more providers, we simply run an `if`/`switch` case and cast accordingly.